teacher.rb

```diff
@@ -1,10 +1,25 @@
-module Teacher
-  def initialize(specialization, age, parent_permission: true, name: 'Unknown')
-    super(name, age, parent_permission)
+require_relative './person'
+class Teacher < Person
+  @@teachers = []
+  def initialize(specialization, age, name, profession)
+    super(age, name, profession)
     @specialization = specialization
+    @@teachers.push(
+      {
+        'name' => name,
+        'specialization' => specialization,
+        'age' => age
+      }
+    )
   end
 
   def can_use_service?
     true
   end
+
+  def teachers_list
+    @@teachers.each_with_index do |teacher, idx|
+      puts "#{idx} || name: #{teacher['name']}, age: #{teacher['age']}, specialization: #{teacher['specialization']}"
+    end
+  end
```

Comment:
- [ ] This should go to the App class where it belongs. It is not behavior of a Rental object.